### PR TITLE
Disable archive balances by env

### PR DESCRIPTION
**Pull request by sl1depengwyn**
## Changelog
Add `ETHEREUM_JSONRPC_DISABLE_ARCHIVE_BALANCES` env var that filters requests and query node only if the block quantity is "latest"

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [x] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [x] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Features
 - [#5968](https://github.com/blockscout/blockscout/pull/5968) - Add call type in the response of txlistinternal API method
 - [#5860](https://github.com/blockscout/blockscout/pull/5860) - Integrate rust verifier micro-service ([blockscout-rs/verifier](https://github.com/blockscout/blockscout-rs/tree/main/verification))
+- [#6001](https://github.com/blockscout/blockscout/pull/6001) - Add ETHEREUM_JSONRPC_DISABLE_ARCHIVE_BALANCES env var that fiters requests and query node only if the block quantity is "latest"
```
Comment (vbaranov) [inline CHANGELOG.md:6]: typo: "fiters"

**Review by vbaranov — COMMENTED** [review]
@sl1depengwyn could you please add test when env variable is set?

File: CHANGELOG.md (hunk 1 above)
Comment (sl1depengwyn) [reply CHANGELOG.md]: fixed

**Comment by sl1depengwyn** [comment]
> @sl1depengwyn could you please add test when env variable is set?

done

https://github.com/blockscout/blockscout/blob/510e7f52eb0e1b69f82744dbb2936608c376b817/apps/ethereum_jsonrpc/test/ethereum_jsonrpc_test.exs#L917-L977

**Review by vbaranov — APPROVED** [review]

**Pull request merged**